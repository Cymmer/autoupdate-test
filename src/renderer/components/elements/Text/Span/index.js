import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';
import styles from '../styles.module.scss';

import textTypes from '../constants/textTypes';

const Span = ({ children, className, type, colorClass }) => (
  <span
    className={cn(className, {
      [styles[`Text___${colorClass}`]]: colorClass !== null,
      [styles[`Text___${type}`]]: type !== null,
    })}
  >
    {children}
  </span>
);

Span.defaultProps = {
  className: null,
  children: null,
  colorClass: null,
  type: null,
};

Span.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  colorClass: PropTypes.oneOf([
    GLOBALS.COLOR_CLASSES.NEUTRAL['900'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['800'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['700'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['600'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['500'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['400'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['300'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['200'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['100'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['50'],
    GLOBALS.COLOR_CLASSES.NEUTRAL['0'],
    GLOBALS.COLOR_CLASSES.BLUE['500'],
    GLOBALS.COLOR_CLASSES.BLUE['300'],
    GLOBALS.COLOR_CLASSES.BLUE['100'],
    GLOBALS.COLOR_CLASSES.RED['500'],
    GLOBALS.COLOR_CLASSES.RED['300'],
    GLOBALS.COLOR_CLASSES.RED['100'],
    GLOBALS.COLOR_CLASSES.GREEN['500'],
    GLOBALS.COLOR_CLASSES.GREEN['300'],
    GLOBALS.COLOR_CLASSES.GREEN['100'],
    GLOBALS.COLOR_CLASSES.YELLOW['400'],
  ]),
  type: PropTypes.oneOf([
    textTypes.HEADING.XXL,
    textTypes.HEADING.XL,
    textTypes.HEADING.LG,
    textTypes.HEADING.MD,
    textTypes.HEADING.SM,
    textTypes.HEADING.XS,
    textTypes.HEADING.XXS,
    textTypes.HEADING.XXXS,
    textTypes.BODY.LG,
    textTypes.BODY.MD,
    textTypes.BODY.SM,
    textTypes.BODY.XS,
    textTypes.STRONG.LG,
    textTypes.STRONG.MD,
    textTypes.STRONG.SM,
    textTypes.STRONG.XS,
    textTypes.DATA.LG,
    textTypes.DATA.MD,
    textTypes.DATA.SM,
    textTypes.DATA.XS,
    textTypes.CODE.MD,
    textTypes.CODE.SM,
  ]),
};

export default Span;
