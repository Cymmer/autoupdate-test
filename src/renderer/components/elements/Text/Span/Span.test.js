import React from 'react';
import { shallow } from 'enzyme';
import GLOBALS from 'codechum-app-globals';
import Span from './index';
import textTypes from '../constants/textTypes';

describe('RENDERING - Text > Span', () => {
  it('should render with default props', () => {
    const component = shallow(<Span />);
    expect(component).toMatchSnapshot();
  });

  it('should render its children', () => {
    const component = shallow(<Span>Test</Span>);
    expect(component).toMatchSnapshot();
  });

  it('should set its passed colorClass in its classname', () => {
    let component = shallow(
      <Span colorClass={GLOBALS.COLOR_CLASSES.BLUE['100']}>Test</Span>
    );
    expect(component).toMatchSnapshot();

    component = shallow(
      <Span colorClass={GLOBALS.COLOR_CLASSES.RED['500']}>Test</Span>
    );
    expect(component).toMatchSnapshot();
  });

  it('should set its passed type in its classname', () => {
    let component = shallow(<Span type={textTypes.HEADING.XXL}>Test</Span>);
    expect(component).toMatchSnapshot();

    component = shallow(<Span type={textTypes.BODY.LG}>Test</Span>);
    expect(component).toMatchSnapshot();
  });
});
