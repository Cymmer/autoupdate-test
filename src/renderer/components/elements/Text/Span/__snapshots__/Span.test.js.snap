// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Text > Span should render its children 1`] = `
<span
  className=""
>
  Test
</span>
`;

exports[`RENDERING - Text > Span should render with default props 1`] = `
<span
  className=""
/>
`;

exports[`RENDERING - Text > Span should set its passed colorClass in its classname 1`] = `
<span
  className="Text___b___100"
>
  Test
</span>
`;

exports[`RENDERING - Text > Span should set its passed colorClass in its classname 2`] = `
<span
  className="Text___r___500"
>
  Test
</span>
`;

exports[`RENDERING - Text > Span should set its passed type in its classname 1`] = `
<span
  className="Text___heading___xxl"
>
  Test
</span>
`;

exports[`RENDERING - Text > Span should set its passed type in its classname 2`] = `
<span
  className="Text___body___lg"
>
  Test
</span>
`;
